addons/website_version/models/ir_ui_view.py

```diff
@@ -0,0 +1,146 @@
+        return self._read_template(cr, uid, view_id, context=context)
+
+    def clear_cache(self):
+        self._read_template.clear_cache(self)
+
+    #To take the right inheriting views
+    @api.model
+    def get_inheriting_views_arch(self, view_id, model):
+        arch = super(view, self).get_inheriting_views_arch(view_id, model)
+        vw = self.browse(view_id)
+        if not (self.env.context and self.env.context.get('website_id') and vw.type == 'qweb'):
+            return arch
+        #right_ids to collect the right ids according to the key
+        right_ids = {}
+        priority = {}
+        #To create a dico called view_arch where v(key) is the id and a the arch(value)
+        view_arch = dict([(v, a) for a, v in arch])
+        keys = self.browse(view_arch.keys()).read(['key', 'version_id', 'website_id'])
+        for k in keys:
+            #The view to take depends of the context
+            if self.env.context.get('version_id'):
+                #priority:1 take the view which is in the same version
+                if k['version_id'] and k['version_id'][0] == self.env.context.get('version_id'):
+                    right_ids[k['key']] = k['id']
+                    priority[k['key']] = 3
+                #priority:2 take the view which is just in the same website
+                elif k['version_id'] is False and k['website_id'] and k['website_id'][0] == self.env.context.get('website_id'):
+                    if not priority.get(k['key']) or priority.get(k['key']) < 3:
+                        right_ids[k['key']] = k['id']
+                        priority[k['key']] = 2
+                #priority:3 take the original view
+                elif k['version_id'] is False and k['website_id'] is False:
+                    if not priority.get(k['key']) or priority.get(k['key']) < 2:
+                        right_ids[k['key']] = k['id']
+                        priority[k['key']] = 1
+            else:
+                #priority:1 take the view which is just in the same website
+                if k['version_id'] is False and k['website_id'] and k['website_id'][0] == self.env.context.get('website_id'):
+                    right_ids[k['key']] = k['id']
+                    priority[k['key']] = 2
+                #priority:2 take the original view
+                elif k['version_id'] is False and k['website_id'] is False:
+                    if not priority.get(k['key']) or priority.get(k['key']) < 2:
+                        right_ids[k['key']] = k['id']
+                        priority[k['key']] = 1
+        return [x for x in arch if x[1] in right_ids.values()]
+        
+    #To active or desactive the right views according to the key
+    def toggle(self, cr, uid, ids, context=None):
+        """ Switches between enabled and disabled statuses
+        """
+        for view in self.browse(cr, uid, ids, context=dict(context or {}, active_test=False)):
```

Comment:
I think you can achieve the same result with two search and perform write in batch, using the new API

def toggle(self):
  self.with_context(active_test=False).search([('key', 'in', self.mapped('key'), ('active', '=', True]).write({'active': False})
  self.with_context(active_test=False).search([('key', 'in', self.mapped('key'), ('active', '=', False]).write({'active': True})